ibis/common/geospatial.py

```diff
@@ -19,22 +19,38 @@ def _format_point_value(value: Iterable) -> str:
     return ' '.join(str(v) for v in value)
 
 
-def _format_linestring_value(value: Iterable) -> str:
+def _format_linestring_value(value: Iterable, nested=False) -> str:
     """Convert a iterable with a linestring to text."""
-    return ', '.join(_format_point_value(point) for point in value)
+    template = '({})' if nested else '{}'
+    # fix wrong structure
+    if not isinstance(value[0], (tuple, list)):
+        value = (value,)
+    return template.format(
+        ', '.join(
+            _format_point_value(point) for point in value
+        )
+    )
 
 
-def _format_polygon_value(value: Iterable) -> str:
+def _format_polygon_value(value: Iterable, nested=False) -> str:
     """Convert a iterable with a polygon to text."""
-    return ', '.join(
-        '({})'.format(_format_linestring_value(line)) for line in value
+    template = '({})' if nested else '{}'
+    # fix wrong structure
+    if not isinstance(value[0][0], (tuple, list)):
+        value = (value,)
+    return template.format(
+        ', '.join(
+            '{}'.format(_format_linestring_value(line, nested=True))
+            for line in value
+        )
     )
 
 
 def _format_multipolygon_value(value: Iterable) -> str:
     """Convert a iterable with a multipolygon to text."""
     return ', '.join(
-        '({})'.format(_format_polygon_value(polygon)) for polygon in value
+        '{}'.format(_format_polygon_value(polygon, nested=True))
```

Comment:
`'{}'.format(x)` is the same as calling `str(x)`, but I don't even think you need to do that here, can't you just call `_format_polygon_value(polygon, nested=True)` and remove the outer `'{}'.format` call?

Reply:
thanks for catch that up! I will fix that!